core/src/main/java/org/apache/iceberg/encryption/Ciphers.java

```diff
@@ -67,7 +83,7 @@ public byte[] encrypt(byte[] plainText, byte[] aad) {
         if (null != aad) {
           cipher.updateAAD(aad);
         }
-        cipher.doFinal(plainText, 0, plainText.length, cipherText, NONCE_LENGTH);
+        cipher.doFinal(plaintext, plaintextOffset, plaintextLength, cipherText, NONCE_LENGTH);
```

Comment:
Does the call to `getInstance` in the constructor guarantee that the `Cipher` instance is not shared? I'm guessing that it must if it has state (mode, key, etc.) but I don't see it guaranteed anywhere.

Reply:
Should this check that the number of bytes returned by `doFinal` is correct?

Reply:
Can you confirm that `doFinal` also writes the GCM tag to the output buffer?

Reply:
> Does the call to getInstance in the constructor guarantee that the Cipher instance is not shared? 

The `AesGcmEncryptor` and `AesGcmDecryptor` are designed for a single thread use.

> Should this check that the number of bytes returned by doFinal is correct?

Sure, we can check this.

> Can you confirm that doFinal also writes the GCM tag to the output buffer?

From the `Cipher.doFinal` javadoc: 
 _If an AEAD mode such as GCM/CCM is being used, the authentication tag is appended in the case of encryption, or verified in the case of decryption._